CMakeLists.txt

```diff
@@ -62,6 +62,7 @@ option(URIPARSER_BUILD_TOOLS "Build tools (e.g. CLI \"uriparse\")" ON)
 option(URIPARSER_BUILD_CHAR "Build code supporting data type 'char'" ON)
 option(URIPARSER_BUILD_WCHAR_T "Build code supporting data type 'wchar_t'" ON)
 set(URIPARSER_MSVC_RUNTIME "" CACHE STRING "Use of specific runtime library (/MT /MTd /MD /MDd) with MSVC")
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
uriparser has a pile of non-public symbols that would be exposed this way, right?
Is this needed for MSVC only? Is selective use of `__declspec(dllexport)` and `__declspec(dllimport)` the alternative approach here? My understanding is that during library compilation we'd need `__declspec(dllexport)` while during application compilation we'd need `__declspec(dllimport)` for the same symbols. I guess `__declspec(dllexport)` would best go to the related `.c` files while `__declspec(dllimport)` would go to .h files? Does that sounds about right? I could make a pull request with that selective approach for you to review. What do you think?

Reply:
Sure.

Reply:
That PR has come to life at #60 now